### PR TITLE
changes made for Longcovid and Monkeypox data

### PR DESCRIPTION
**Pull request by urvi-occ**
The long covid working group is working with OCC towards getting Long covid and monkeypox data into Pandemic response commons. This PR consist of DD changes for PRC to accommodate these data and to align the data dictionary to accommodate new fields for long covid and monkeypox data. The long covid working group has put together this documentation in which we have information to get patients' data to do analysis on long covid. [Document](https://docs.google.com/document/d/1zOU7g1pUgpGpH5GesOfcYDClZ3F074Fg8UsjriDdbmY/edit#heading=h.mep11nve4jer)

This specific PR is focusing on getting the modification to the existing nodes which are subject and observation

- subject node: Add unique_de-id_identifier to the subject node (required)

- observation node:
- Observation_icd10 -  ICD-10 code for encounter 
- Observation_cpt - CPT code for encounter

### CODE REVIEW

File: gdcdictionary/schemas/subject.yaml
Comment (paulineribeyre) [inline gdcdictionary/schemas/subject.yaml:44]: making this a required field is a breaking change for all the existing subject data that doesn't have it

File: gdcdictionary/schemas/subject.yaml
Comment (paulineribeyre) [inline gdcdictionary/schemas/subject.yaml:172]: it's not clear to me from the description what this ID is

File: gdcdictionary/schemas/observation.yaml
Comment (paulineribeyre) [inline gdcdictionary/schemas/observation.yaml:781]:
```suggestion
      zero, one or more ICD-10 codes for encounter
```
if they are stored in an array, then they are not a comma-separated string

File: gdcdictionary/schemas/observation.yaml
Comment (paulineribeyre) [inline gdcdictionary/schemas/observation.yaml:788]: same here

File: gdcdictionary/schemas/subject.yaml
Comment (urvi-occ) [reply gdcdictionary/schemas/subject.yaml:172]: The unique_de-id_identifier is a unique identifier for any subject/patient. This is unique per subject and it is continous all through system.

File: gdcdictionary/schemas/subject.yaml
Comment (urvi-occ) [reply gdcdictionary/schemas/subject.yaml:44]: It was asked to be required by long covid group, is there any other way to make in required ?

File: gdcdictionary/schemas/observation.yaml
Comment (urvi-occ) [reply gdcdictionary/schemas/observation.yaml]: code updated

File: gdcdictionary/schemas/subject.yaml
Comment (paulineribeyre) [reply gdcdictionary/schemas/subject.yaml:172]: can we just make it the `submitter_id` then? 🙂 that's the conventional unique ID across Gen3

File: gdcdictionary/schemas/subject.yaml
Comment (paulineribeyre) [reply gdcdictionary/schemas/subject.yaml:44]: Not in the dictionary - you will need to make sure that it is always specified on the data ingestion side

Comment (urvi-occ) [reply gdcdictionary/schemas/subject.yaml:44]: oh okay, I can do that. Let me make that change and commit

File: gdcdictionary/schemas/subject.yaml
Comment (urvi-occ) [reply gdcdictionary/schemas/subject.yaml:172]: Actually, this id is provided by the healthcare organization and we would be utilizing it for research. I would recommend separating the submitter id from this so it doesn't impact other projects

File: gdcdictionary/schemas/subject.yaml
Comment (urvi-occ) [reply gdcdictionary/schemas/subject.yaml]: I have made the commit with this change.

File: gdcdictionary/schemas/subject.yaml
Comment (urvi-occ) [reply gdcdictionary/schemas/subject.yaml]: Actually, this id is provided by the healthcare organization and we would be utilizing it for research. I would recommend separating the submitter id from this submitter id as we are currently using submitter_id to get the name of the org and get the submission date of that particular subject as shown below.

Screen Shot 2023-01-25 at 1 52 37 AM

Comment (paulineribeyre) [reply gdcdictionary/schemas/subject.yaml]: it's your call so i will approve the PR to unblock you, but this sounds to me like the org and date should be stored in their own properties if you think they are useful information, and the unique ID should be stored in the existing property that is meant for IDs that are unique across the project => submitter_id

**Review by paulineribeyre — APPROVED** [review]

File: gdcdictionary/schemas/subject.yaml
Comment (urvi-occ) [reply gdcdictionary/schemas/subject.yaml]: That's a good insight. Let me talk to team internally if we can take this approach in the next iteration. Can you approve this PR whenever you can?

File: gdcdictionary/schemas/observation.yaml
Comment (paulineribeyre) [inline gdcdictionary/schemas/observation.yaml:791]: i think the indentation here is causing the tests failures

File: gdcdictionary/schemas/subject.yaml
Comment (paulineribeyre) [reply gdcdictionary/schemas/subject.yaml]: sure
i already approved it, the tests didn't pass though

File: gdcdictionary/schemas/observation.yaml
Comment (urvi-occ) [reply gdcdictionary/schemas/observation.yaml]: I am sorry about that. I have updated the code and made commit.

File: gdcdictionary/schemas/subject.yaml
Comment (paulineribeyre) [inline gdcdictionary/schemas/subject.yaml:169]: looks like property names cannot contain `-`:
```
Validating 'Subject'
ERROR
'unique_de-id_identifier' does not match any of the regexes: '^[_a-zA-Z0-9]*$'
```

Comment (urvi-occ) [reply gdcdictionary/schemas/subject.yaml]: changed `unique_de-id_identifier` to `unique_de_id_identifier`

**Review by paulineribeyre — APPROVED** [review]

**Pull request merged**
